### ISSUE TITLE
Obscure errors when incorectly configured

### ISSUE THREAD

**Issue by ignaloidas** [elastic/elasticsearch-py#1386]
**`elasticsearch-py` version (`elasticsearch.__versionstr__`)**: 7.9.1

**Description of the problem including expected versus actual behavior**:
I've set Elasticsearch in my code like this:
```python
es = Elasticsearch(
    hosts=[{"host":"https://example.com", "port":5001}],
    http_auth = custom_auth,
    use_ssl=True,
    verify_certs=True,
    connection_class=RequestsHttpConnection,
)
```
All requests failed with `[Errno -3] Temporary failure in name resolution`. After debugging a bunch, I figured out that because my entry in `hosts` is a dictionary, the "https://" is not removed from the host, and another one is added because I used `use_ssl=True`, which leads to host passed to requests be `https://https//example.com`, which leads to it trying to lookup `https`, which of course fails.

**Comment by sethmlarson** [comment]
Yeah unfortunately we can't cover all the bases for input validation. If you'd like to add a change that raises a configuration error when receiving `://` in `host` I'd accept that.

**Comment by sethmlarson** [comment]
This has been resolved and will be released in 8.0.
